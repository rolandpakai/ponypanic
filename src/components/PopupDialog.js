import { useState } from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';


const PopupDialog = (props) => {
  //const [open, setOpen] = useState(false);
  const { dialogContentText, buttonOkText, buttonCancelText, open, okHandler, cancelHandler } = {...props};

  return (
    <Dialog
    open={open}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
    PaperProps={{
      style: {
          textAlign: 'center',
      },
   }}
  >
    <DialogContent>
      <DialogContentText id="alert-dialog-description">
        {dialogContentText}
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={cancelHandler}>{buttonCancelText}</Button>
      <Button onClick={okHandler} autoFocus>{buttonOkText}</Button>
    </DialogActions>
  </Dialog>
  )
}

export default PopupDialog;

