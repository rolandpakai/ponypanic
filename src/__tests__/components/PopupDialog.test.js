import { render, act } from '@testing-library/react';
import PopupDialog from '../../components/PopupDialog';

describe('PopupDialog Component', () => {
  test('PopupDialog renders', () => {
    const {container} = render(<PopupDialog open={true} />);
    
		expect(container).toMatchSnapshot();
  });
});