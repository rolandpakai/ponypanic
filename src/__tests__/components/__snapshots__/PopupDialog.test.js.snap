// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupDialog Component PopupDialog renders 1`] = `
<div
  aria-hidden="true"
/>
`;
